downloads/swan-lake-release-notes/swan-lake-beta4/RELEASE_NOTE.md

```diff
@@ -0,0 +1,827 @@
+        io : print("Ballerina"); // compilation error: intervening whitespaces are not allowed in a qualified identifier.
+    }
+    ```
+
+    With this, `x ? a : b:c` will now be parsed as `x ? a : (b:c)` since the colon with spaces is interpreted only as part of a conditional expression.
+
+- Fixed a bug that resulted in hash collisions not being handled correctly in `table` values.
+
+    ```ballerina
+    import ballerina/io;
+    
+    public function main() {
+        table<record {readonly int? k;}> key(k) t = table [];
+        t.add({k: 0});
+        io:println(t.hasKey(()));
+    }
+    ```
+
+    The above code snippet, which previously printed `true` will now print `false`.
+
+- Disallowed object type inclusions with an object that has private fields or members.
+
+    ```ballerina
+    class Person {
+        string firstName;
+        string lastName;
+        private string dob;
+
+        function init(string firstName, string lastName, string dob) {
+            self.firstName = firstName;
+            self.lastName = lastName;
+            self.dob = dob;
+        }
+
+        private function getName() returns string => self.firstName + self.lastName;
+    }
+    
+    class Employee {
+        *Person; // Will now result in an error.
+        int id;
+
+        function init(string firstName, string lastName, string dob, int id) {
+            self.firstName = firstName;
+            self.lastName = lastName;
+            self.dob = dob;       
+            self.id = id;
+        }
+
+        private function getName() returns string => self.firstName;	
+    }
+    ```
+
+- Fixed a bug that resulted in compilation errors not being emitted for invalid `xml` template expressions.
+
+    ```ballerina
+    xml x = xml `</>`; // Will now result in an error.
+    ```
+
+- Fixed a bug that resulted in compilation errors not being emitted for duplicate fields written with escape sequences in the mapping constructor.
+
+    ```ballerina
+    map<any> x = {a\\: 454, "a\\": false}; // Will now result in an error.
+    ```
+
+- Updated the `xml:createElement` to accept the attribute map as the second argument.
+
+    ```ballerina
+    xml:Element jo = xml:createElement("name", {id: "1234"}, xml `Jo`);
+    ```
+
+- Updated the `xml:get` function’s return type to return the exact type `T` when the `xml` sequence is of type `xml<T>`.
+
+    ```ballerina
+    xml<xml:Element> employees = xml `<e1><name>Jo</name></e1><e2><name>Mary</name></e2>`;
+    xml:Element employee1 = employees.get(0);// Now allowed.
+    ```
+
+- Updated the `table:map` function’s function argument `func` and the return type to work with subtypes of mapping types instead of any type.
+
+    ```ballerina
+    table<record {int id; string name;}> tb = table [
+    {id: 1234, name: "Jo"},
+    {id: 2345, name: "May"}
+    ];
+
+    var idTable = tb.map(function (record {int id; string name;} r) returns int { // No longer allowed.
+    return r.id;
+    });
```

Comment:
This code block is not properly formatted

Reply:
Fixed.